fabric-datafixer-v0/src/main/java/net/fabricmc/fabric/impl/datafixer/FabricDataFixerInvoker.java

```diff
@@ -0,0 +1,84 @@
+package net.fabricmc.fabric.impl.datafixer;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.mojang.datafixers.DataFixer;
+import com.mojang.datafixers.Dynamic;
+
+import net.fabricmc.fabric.api.util.NbtType;
+import net.fabricmc.fabric.impl.datafixer.FabricDataFixerInvoker.DataFixerEntry;
+import net.minecraft.datafixers.DataFixTypes;
+import net.minecraft.datafixers.NbtOps;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.Tag;
+import net.minecraft.util.crash.CrashException;
+import net.minecraft.util.crash.CrashReport;
+import net.minecraft.util.crash.CrashReportSection;
+
+public class FabricDataFixerInvoker {
+    
+    Map<String, DataFixerEntry> MOD_FIXERS = new HashMap<String, DataFixerEntry>();
+    
+    private FabricDataFixerInvoker() {}
+    
+    public static final FabricDataFixerInvoker INSTANCE = new FabricDataFixerInvoker();;
+    
+    public void registerFixer(String modid, int runtimeDataVersion, DataFixer datafixer) {
+        MOD_FIXERS.put(modid, new DataFixerEntry(datafixer, runtimeDataVersion));
+    }
+    
+    public static CompoundTag updateWithAllFixers(DataFixer dataFixer_1, DataFixTypes dataFixTypes_1, CompoundTag compoundTag_1, int dyanamicDataVersion, int runtimeDataVersion) {
+        Iterator<Entry<String, DataFixerEntry>> iterator = INSTANCE.MOD_FIXERS.entrySet().iterator();
+        CompoundTag currentTag = compoundTag_1;
+        
+        while(iterator.hasNext()) {
```

Comment:
Use foreach loop

Reply:
Impossible because dataFixer.update(args...) returns a new CompoundTag and I have to replace local var with new tag as you see further down in iterator. If I were to use a foreach loop the compiler would fail because it wouldn't be final or effectively final.

Reply:
An actual foreach loop like this should not require final variables:
```java
for (Entry<String, DataFixerEntry> entry : INSTANCE.MOD_FIXERS.entrySet()) { ... }
```

Reply:
Yeah now I see, I was thinking of the lambda forEach stream loop and originally tried to avoid it because of streams performance.